tce/opset/base/implicit_lsu.cc

```diff
@@ -0,0 +1,300 @@
+/*
+    Copyright (c) 2002-2020 Tampere University.
+
+    This file is part of TTA-Based Codesign Environment (TCE).
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+ */
+/**
+ * @file implicit_lsu.cc
+ *
+ * Implicit LSU operation definitions.
+ *
+ * @author Kanishkan Vadivel (k.vadivel-no.spam-tue.nl)
+ * @author Barry de Bruin (e.d.bruin-no.spam-tue.nl)
+ */
+
+#include "OSAL.hh"
+#include "OperationGlobals.hh"
+
+#define RUNTIME_ERROR_WITH_INT(MESSAGE, DATA) {\
```

Comment:
Can you make a generic RUNTIME_ERROR with a varlength so we do not need this special macros for different parameters?

Reply:
moved to single macro.